independent-projects/resteasy-reactive/server/vertx/src/main/java/org/jboss/resteasy/reactive/server/vertx/AppendBuffer.java

```diff
@@ -1,53 +1,81 @@
 package org.jboss.resteasy.reactive.server.vertx;
 
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
 import java.util.ArrayDeque;
 import java.util.Objects;
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.ByteBufAllocator;
 import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.PooledByteBufAllocator;
+import io.netty.util.internal.PlatformDependent;
+import io.vertx.core.buffer.impl.PartialPooledByteBufAllocator;
+import io.vertx.core.buffer.impl.VertxByteBufAllocator;
 
 /**
  * It is a bounded (direct) buffer container that can keep on accepting data till {@link #capacity} is exhausted.<br>
  * In order to keep appending on it, it can {@link #clear} and consolidate its content as a {@link ByteBuf}.
  */
 final class AppendBuffer {
-    private final ByteBufAllocator allocator;
+
+    private static final MethodHandle virtualMh = PlatformDependent.javaVersion() >= 21 ? findVirtualMH() : null;
```

Comment:
maybe we have a better way which work with both JVM and native image here @geoand ? IDK

Reply:
We'll probably need a GraalVM substitution for this

Reply:
Consider that vertx does the same exact thing at https://github.com/eclipse-vertx/vert.x/blob/adbe97600cc6215f15ce2fac629c89f93618ca8f/src/main/java/io/vertx/core/json/jackson/HybridJacksonPool.java#L37 but without checking the JDK version (that's not needed I think) - which means that *if* we use the HybridJacksonPool it's going to do the same as I do here.

Reply:
It might even work OOTB in the latest GraalVM versions, I am not sure really